src/sentry/notifications/platform/types.py

```diff
@@ -21,21 +15,12 @@ class NotificationProviderKey(StrEnum):
     DISCORD = ExternalProviderEnum.DISCORD
 
 
-type NotificationRenderable = DiscordRenderable | EmailRenderable | SlackRenderable | MSTeamsRenderable
-"""
-Union type of all notification known renderables. When a new provider inherits the
-Provides stricter typing for permitted notification providers added to the registry.
-"""
-
-
-NotificationRenderableT = TypeVar("NotificationRenderableT", bound=NotificationRenderable)
+NotificationRenderableT = TypeVar("NotificationRenderableT")
 """
 A renderable object that is understood by the notification provider.
 For example, Email might expect HTML, or raw text; Slack might expect a JSON Block Kit object.
 """
-NotificationRenderableT_co = TypeVar(
-    "NotificationRenderableT_co", bound=NotificationRenderable, covariant=True
-)
+NotificationRenderableT_co = TypeVar("NotificationRenderableT_co", covariant=True)
 """
 A covariant type of NotificationRenderableT, used in the renderer to pass the correct type through
 to the render method.
```

Comment:
we are on python3.12+ so you can use the new syntax with automatic generics and then you do not need these at all